centrallix/osdrivers/objdrv_xml.c

```diff
@@ -469,13 +471,13 @@ xml_internal_ReadDoc(pObject obj)
 	/** Not in cache, or cache was stale **/
 	if (!pCache)	
 	    {
-	    if(XML_DEBUG) printf("couldn't find %s in cache\n",path);
+	    if(XML_DEBUG) printf("couldn't find %s in cache\n",hash);
 	    pCache=(pXmlCacheObj)nmMalloc(sizeof(XmlCacheObj));
 	    if(!pCache) return NULL;
 	    memset(pCache,0,sizeof(XmlCacheObj));
 	    if (objGetAttrValue(obj->Prev, "last_modification", DATA_T_DATETIME, POD(&pDT)) == 0)
 		pCache->lastmod.Value = pDT->Value;
-	    strtcpy(pCache->Pathname, path, sizeof(pCache->Pathname));
+	    strtcpy(pCache->Pathname, hash, sizeof(pCache->Pathname));
```

Comment:
Ensure that pXmlCacheObj -> Pathname is the same length as 'hash', otherwise you could get false cache misses because the cache's idea of the serialization is different from the lookup computed value.  Perhaps use a #define in order to ensure that those lengths match.

Reply:
I think this latest change should address the problem. Let me know what you think

Reply:
A couple of minor notes (not blocking this PR) - in the GetDCHash call, use sizeof(hash) rather than XML_HASH_SIZE, and in the #define where you have a computed value, put parens around the expression so you don't get unexpected interactions such as 2*XML_HASH_SIZE (hypothetical) not doing what one might expect.